### PR TITLE
fix(readme): Add min golang version

### PR DESCRIPTION
**Pull request by davidffrench**

### CODE REVIEW

**Comment by 2uasimojo** [comment]
Should we update [go.mod](https://github.com/kcp-dev/kcp/blob/fd28572bb60d5b802284185e987780b60bcc73fe/go.mod#L3) as well?

File: README.md [hunk 1]
```diff
@@ -84,10 +84,10 @@ However, `kcp` is not an acronym.
```
Comment (2uasimojo) [inline README.md:84]: While you're in here, would you mind fixing the typo on [L68](https://github.com/kcp-dev/kcp/pull/52/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R68)? s/reconiling/reconciling/

Comment (davidffrench) [reply README.md:84]: good spotting, fixed now

**Review by imjasonh — APPROVED** [review]

**Comment by 2uasimojo** [comment]
> Should we update [go.mod](https://github.com/kcp-dev/kcp/blob/fd28572bb60d5b802284185e987780b60bcc73fe/go.mod#L3) as well?

Proposed via #58.

**Pull request merged**